addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/handler/EchoHandler.java

```diff
@@ -1137,4 +1242,67 @@ public void handlePushCommand(String command, String payload) {
                 }
         }
     }
+
+    public void updateNotifications(ZonedDateTime currentTime, ZonedDateTime now,
```

Comment:
The `EchoHandler` class is also quite large and you are adding another long, public method to it. Could all of the notification related stuff go into its own class?

Reply:
Not yet. The refactoring is planned to do in the next 2-3 month.